Kernel/Process.h

```diff
@@ -531,7 +533,7 @@ class Process
     void delete_perf_events_buffer();
 
     KResult do_exec(NonnullRefPtr<FileDescription> main_program_description, Vector<String> arguments, Vector<String> environment, RefPtr<FileDescription> interpreter_description, Thread*& new_main_thread, u32& prev_flags, const Elf32_Ehdr& main_program_header);
-    KResultOr<ssize_t> do_write(FileDescription&, const UserOrKernelBuffer&, size_t);
+    KResultOr<ssize_t> do_write(FileDescription&, const UserOrKernelBuffer&, size_t, bool pw = false, off_t offset = 0);
```

Comment:
not sure if it's allowed inside kernel but maybe instead of `bool` and `off_t` you could use an `AK::Optional<off_t>`, pass and empty optional for write and a non empty optional for pwrite.